app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -669,6 +696,196 @@ class OnboardingFragmentTest {
+  @Test
+  fun testOnboardingFragment_onboardingV2Enabled_tabletLandscape_screenIsCorrectlyDisplayed() {
+    setUpTestWithOnboardingV2Enabled()
+
+    launch(OnboardingActivity::class.java).use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_language_title)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_subtitle)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_text)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_label)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_dropdown_background)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_explanation)).check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_language_lets_go_button)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testOnboardingFragment_onboardingV2Enabled_allIcons_haveCorrectContentDescriptions() {
+    setUpTestWithOnboardingV2Enabled()
+
+    launch(OnboardingActivity::class.java).use {
+      onView(withId(R.id.onboarding_language_dropdown_arrow)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_language_dropdown_arrow_icon_description
+          )
+        )
+      )
+      onView(withId(R.id.onboarding_app_language_image)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_otter_content_description
+          )
+        )
+      )
+      onView(withId(R.id.onboarding_language_dropdown_icon)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_language_dropdown_icon_description
+          )
+        )
+      )
+    }
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testFragment_onboardingV2Enabled_mobileLandscape_allIcons_haveCorrectContentDescriptions() {
+    setUpTestWithOnboardingV2Enabled()
+
+    launch(OnboardingActivity::class.java).use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_language_dropdown_arrow)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_language_dropdown_arrow_icon_description
+          )
+        )
+      )
+      onView(withId(R.id.onboarding_app_language_image)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_otter_content_description
+          )
+        )
+      )
+      onView(withId(R.id.onboarding_language_dropdown_icon)).check(
+        matches(
+          withContentDescription(
+            R.string.onboarding_language_dropdown_icon_description
+          )
+        )
+      )
+    }
+  }
+
+  @Config(qualifiers = "sw600dp-port")
+  @Test
+  fun testFragment_onboardingV2Enabled_mobilePortrait_allIcons_haveCorrectContentDescriptions() {
```

Comment:
Isn't this tabletPortrait?

Reply:
Yes it was - I have since consolidated the ContentDescriptions checks with the rest of the views.